### PR TITLE
PLU-70: [TILES-11] frontend: Table editor

### PR DESCRIPTION
**Pull request by pregnantboy**
image

Features
- Fetch all rows
- Update single row
- Virtualised rows
- Keyboard focus manipulation

To test:
- manually create columns in DB
- Seed db with script

### CODE REVIEW

File: packages/frontend/src/pages/Tile/index.tsx [hunk 1]
```diff
@@ -1,10 +1,51 @@
...
+import { GET_TABLE } from 'graphql/queries/get-table'
+
+import Table from './components/Table'
 
 export default function Tile(): JSX.Element {
   const { tableId } = useParams<{ tableId: string }>()
 
+  const { data: getTableData, loading: getTableLoading } = useQuery<{
+    getTable: ITableMetadata
+  }>(GET_TABLE, {
```
Comment (ogp-weeloong) [inline packages/frontend/src/pages/Tile/index.tsx:17]: nit: we can combine into a single query TBH, e.g.:

```
getTable(tableId: $tableId) {
  id
  name
  columns {
    id
    name
  }
}
getAllRows(tableId: $tableId) {
  rowId
  data
}
```

or even just adding a rows field to table. that said I can see the advantages of modularizing the queries so this is just a nit.

File: packages/frontend/src/pages/Tile/index.tsx [hunk 2]
```diff
@@ -1,10 +1,51 @@
...
+  const { data: getAllRowsData, loading: getAllRowsLoading } = useQuery<{
+    getAllRows: ITableRow[]
+  }>(GET_ALL_ROWS, {
+    variables: {
+      tableId,
+    },
+  })
+
+  if (
+    !getTableData?.getTable ||
```
Comment (ogp-weeloong) [inline packages/frontend/src/pages/Tile/index.tsx:32]: nit: I try to sell koyok to you all on friday about convering to useSuspenseQuery and useBackgroundQuery :P

File: packages/frontend/src/pages/Tile/index.tsx [hunk 3]
```diff
@@ -1,10 +1,51 @@
...
+    !getAllRowsData?.getAllRows ||
+    getTableLoading ||
+    getAllRowsLoading
+  ) {
+    return (
+      <Center height="100vh">
+        <Spinner size="xl" thickness="4px" color="primary.500" margin="auto" />
```
Comment (ogp-weeloong) [inline packages/frontend/src/pages/Tile/index.tsx:39]: nit: OGP spinner instead?

**Review by ogp-weeloong — APPROVED** [review]
LGTM, tested row editing

File: packages/frontend/src/pages/Tile/index.tsx (hunk 2 above)
Comment (pregnantboy) [reply packages/frontend/src/pages/Tile/index.tsx:32]: useSuspenseQuery does look cleaner

**Pull request merged**